service/src/main/java/bio/terra/axonserver/service/cloud/gcp/CloudStorageService.java

```diff
@@ -0,0 +1,52 @@
+package bio.terra.axonserver.service.cloud.gcp;
+
+import bio.terra.axonserver.service.exception.CloudObjectReadException;
+import bio.terra.axonserver.utils.BoundedByteArrayOutputStream;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.ReadChannel;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.StorageOptions;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/** Service for interacting with Google Cloud Storage */
+public class CloudStorageService {
+
+  private static final int MAX_OBJECT_SIZE = 512 * 1024 * 1024; // 512 MB
+  private static final int MAX_BUFFER_SIZE = 64 * 1024; // 64 KB
```

Comment:
Not sure if this is an improvement, but there's a spring util `DataSize` for working with MB/KB values. It returns `long` so a cast is required.
```suggestion
  private static final int MAX_OBJECT_SIZE = (int) DataSize.ofMegabytes(512).toBytes();
  private static final int MAX_BUFFER_SIZE = (int) DataSize.ofKilobytes(64).toBytes();
```

Reply:
This helps with readability